server/parser.go

```diff
@@ -1175,3 +1180,18 @@ func (c *client) clonePubArg() error {
 		}
 	}
 }
+
+func (ps *parseState) getHeader() http.Header {
```

Comment:
Technically, c.parseState is embedded and not a pointer, so weird to have this receiver on a pointer. But really I wonder if this should not be a receiver for `*client` directly, and you would do `c.getHeader()`.

Alternatively, I was saying that the creation of the header could be in a generic place (or here in parser.go) but actually not tied to the `parseState` in case we need to parse headers in other conditions. In that case we would have to pass msgBuf and pa.hdr as arguments.
We don't need to do that now though.